cmd/openvdc/cmd/copy.go

```diff
@@ -0,0 +1,76 @@
+package cmd
+
+import (
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/axsh/openvdc/api"
+	filecopy "github.com/axsh/openvdc/cmd/openvdc/cmd/copy"
+	"github.com/axsh/openvdc/cmd/openvdc/internal/util"
+	"github.com/spf13/cobra"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+var copyCmd = &cobra.Command{
+	Use:   "copy [File src path] [Instance ID]:[file dest path]",
+	Short: "Copy files to and between instances",
+	Long:  "Copy files to and between instances",
+	Example: `
+	% openvdc copy 1.txt i-xxxxxxx:/tmp/1.txt
+	`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) == 0 {
+			log.Fatalf("Please provide a source path.")
+		}
+		if len(args) == 1 {
+			log.Fatalf("Please provide a destination path.")
+		}
+
+		src := args[0]
+		dest := args[1]
+
+		p := strings.Split(dest, ":")
+		if len(p) < 2 {
+			log.Fatalf("Invalid destination path. Please use this format: [Instance ID]:[file dest path]")
+		}
+
+		instanceID, instanceDir := p[0], p[1]
+		if instanceID == "" {
+			log.Fatalf("Invalid Instance ID")
+		}
+		if instanceDir == "" {
+			log.Fatalf("Invalid destination path")
+		}
+
+		req := &api.CopyRequest{
+			InstanceId: instanceID,
+		}
+
+		var res *api.CopyReply
+
+		err := util.RemoteCall(func(conn *grpc.ClientConn) error {
+			c := api.NewInstanceClient(conn)
+			var err error
+			res, err = c.Copy(context.Background(), req)
+			return err
+		})
+
+		if err != nil {
+			log.WithError(err).Fatal("Disconnected abnormally")
+		}
+
+		client, err := filecopy.NewClient(res)
+		if err != nil {
+			log.WithError(err).Fatal("Failed to create client")
+		}
+
+		err = client.CopyFile(src, instanceDir)
```

Comment:
Does ``instanceDir`` point to the path on the host filesystem? It does not make sense. This feature is to copy the file into the instance filesystem not to the host filesystem.

Reply:
Example:
```./openvdc copy testfile i-0000000001:/tmp/```

instanceDir refers to whichever folder you want to put the file in on the instance filesystem, in this case "/tmp/".

The file ends up in ```"/var/lib/lxc/i-0000000001/rootfs/tmp/"``` not ```"/tmp/"```

Reply:
Miss read the code. ``/usr/bin/scp`` is run by executor's sshd so that data stream is sent to the container.